commands/volumes/volume-create.go

```diff
@@ -43,68 +47,117 @@ func createVolume(msg *volume.VolCreateRequest) (*volume.Volinfo, error) {
 	return vol, nil
 }
 
-func validateVolumeCreate(msg *volume.VolCreateRequest, v *volume.Volinfo) (int, error) {
-	if volume.ExistsFunc(msg.Name) {
-		log.WithField("volume", msg.Name).Error("Volume already exists")
-		return http.StatusBadRequest, errors.ErrVolExists
+func validateVolumeCreate(c *context.Context) error {
+	req, ok := c.Get("req").(*volume.VolCreateRequest)
+	if !ok {
+		return errors.New("failed to get request from context")
 	}
-	httpStatusCode, err := volume.ValidateBrickEntriesFunc(v.Bricks, v.ID, msg.Force)
+
+	if volume.ExistsFunc(req.Name) {
+		c.Log.WithField("volume", req.Name).Error("volume already exists")
+		return gderrors.ErrVolExists
+	}
+
+	vol, err := createVolinfo(req)
 	if err != nil {
-		return httpStatusCode, err
+		return err
 	}
-	return 0, nil
+
+	_, err = volume.ValidateBrickEntriesFunc(vol.Bricks, vol.ID, req.Force)
+	if err != nil {
+		return err
+	}
+
+	// Store volinfo for later usage
+	c.Set("volinfo", vol)
+
+	return nil
 }
 
-func commitVolumeCreate(vol *volume.Volinfo) (int, error) {
+func generateVolfiles(c *context.Context) error {
+	vol, ok := c.Get("volinfo").(*volume.Volinfo)
+	if !ok {
+		return errors.New("failed to get volinfo from context")
```

Comment:
I will, but in a later PR. Dumping errors of different packages into a single `error` package doesn't seem really good to me. I'll do a refactoring of it, and will add this in it.